File: app/src/main/java/de/bikebean/app/db/type/types/sms_parser_types/WifiList.java
package de.bikebean.app.db.type.types.sms_parser_types;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.lang.ref.WeakReference;

import de.bikebean.app.db.settings.settings.Battery;
import de.bikebean.app.db.settings.settings.Wapp;
import de.bikebean.app.db.settings.settings.number_settings.WifiAccessPoints;
import de.bikebean.app.db.sms.Sms;
import de.bikebean.app.db.state.State;
import de.bikebean.app.db.type.types.SmsParserType;
import de.bikebean.app.ui.drawer.log.LogViewModel;

public class WifiList extends SmsParserType {

    private static boolean matches() {
        return wifiMatcher.find(0) && batteryMatcher.find(0);
    }

    public static @Nullable WifiList createIfMatches(
            final @NonNull Sms sms,
            final @NonNull WeakReference<LogViewModel> lv) {
        wifiMatcher = wifiPattern.matcher(sms.getBody());
        batteryMatcher = batteryPattern.matcher(sms.getBody());

        if (matches())
            return new WifiList(sms, lv);

        return null;
    }

    private WifiList(final @NonNull Sms sms, final @NonNull WeakReference<LogViewModel> lv) {
        super(TYPE.WIFI_LIST, sms, lv);

        /* battery value is encoded differently in this case */
        getSettings().add(new WifiAccessPoints(sms, super::getWappWifiAccessPoints));
        getSettings().add(new Wapp(sms, State.WAPP_WIFI_ACCESS_POINTS));
        getSettings().add(new Battery(sms, super::getBattery));
    }
}

File: app/src/main/java/de/bikebean/app/db/type/types/sms_parser_types/WifiList.kt
package de.bikebean.app.db.type.types.sms_parser_types

import de.bikebean.app.db.settings.settings.add_to_list_settings.Battery
import de.bikebean.app.db.settings.settings.add_to_list_settings.Wapp
import de.bikebean.app.db.settings.settings.add_to_list_settings.number_settings.WifiAccessPoints
import de.bikebean.app.db.sms.Sms
import de.bikebean.app.db.state.State
import de.bikebean.app.db.type.types.ParserPatterns.batteryPattern
import de.bikebean.app.db.type.types.ParserPatterns.wifiPattern
import de.bikebean.app.db.type.types.ParserType
import de.bikebean.app.ui.drawer.log.LogViewModel
import java.lang.ref.WeakReference

class WifiList(sms: Sms, lv: WeakReference<LogViewModel>) : ParserType(TYPE.WIFI_LIST, sms, lv) {

    init {
        wifiMatcher = wifiPattern.matcher(sms.body)
        batteryMatcher = batteryPattern.matcher(sms.body)
    }

    override val matchers = listOf(
            wifiMatcher,
            batteryMatcher
    )

    override val settings get() = listOf(
            WifiAccessPoints(sms) { wappWifiAccessPoints },
            Wapp(sms, State.WAPP_WIFI_ACCESS_POINTS),
            /* battery value is encoded differently in this case */
            Battery(sms) { battery }
    )

}